zulipterminal/ui_tools/views.py

```diff
@@ -814,7 +813,13 @@ def calculate_table_widths(contents: PopUpViewTableContent,
         for category, content in contents:
             category_width = max(category_width, len(category))
             for row in content:
-                strip_widths.append(list(map(len, row)))
+                # Measure the longest line if the text is seperated by
+                # newline(s).
+                max_row_lengths = [
+                    len(max(text.split('\n'), key=len))
+                    for text in row
+                ]
+                strip_widths.append(max_row_lengths)
```

Comment:
This looks clearer in this commit than the version with lambdas before! This does seem to fix the issue we saw earlier with this commit, unless the window/width gets super-narrow, which is probably going to be an issue to use ZT in any case.